plugins/wasmembed/wasmembed.cpp

```diff
@@ -0,0 +1,149 @@
+                }
+            }
+        }
+
+        //  IWasmEmbedCallback  ---
+        virtual inline void DBGLOG(const char *format, ...) override
+        {
+            va_list args;
+            va_start(args, format);
+            VALOG(MCdebugInfo, unknownJob, format, args);
+            va_end(args);
+        }
+
+        virtual void throwStringException(int code, const char *format, ...) override
+        {
+            va_list args;
+            va_start(args, format);
+            IException *ret = makeStringExceptionVA(code, format, args);
+            va_end(args);
+            throw ret;
+        }
+
+        virtual const char *resolveManifestPath(const char *leafName) override
+        {
+            if (leafName && *leafName)
+            {
+                ForEachItemIn(idx, manifestModules)
+                {
+                    const char *path = manifestModules.item(idx);
+                    if (endsWith(path, leafName))
+                        return path;
+                }
+            }
+            return nullptr;
+        }
+    };
+    std::shared_ptr<Callbacks> callbacks;
+
+    class WasmEmbedContext : public CInterfaceOf<IEmbedContext>
+    {
+        std::unique_ptr<ISecureEnclave> enclave;
+
+    public:
+        WasmEmbedContext()
+        {
+            enclave = createISecureEnclave();
+        }
+        virtual ~WasmEmbedContext() override
+        {
+        }
+        //  IEmbedContext  ---
+        virtual IEmbedFunctionContext *createFunctionContext(unsigned flags, const char *options) override
+        {
+            return createFunctionContextEx(nullptr, nullptr, flags, options);
+        }
+        virtual IEmbedFunctionContext *createFunctionContextEx(ICodeContext *ctx, const IThorActivityContext *activityContext, unsigned flags, const char *options) override
+        {
+            callbacks->manifestPaths(ctx);
+            return enclave.get();
+        }
+        virtual IEmbedServiceContext *createServiceContext(const char *service, unsigned flags, const char *options) override
+        {
+            throwUnexpected();
+            return nullptr;
+        }
+    };
+
+    extern DECL_EXPORT IEmbedContext *getEmbedContext()
+    {
+        return new WasmEmbedContext();
+    }
+
+    extern DECL_EXPORT void syntaxCheck(size32_t &__lenResult, char *&__result, const char *funcname, size32_t charsBody, const char *body, const char *argNames, const char *compilerOptions, const char *persistOptions)
+    {
+        StringBuffer result;
+        //  MORE - ::syntaxCheck(__lenResult, __result, funcname, charsBody, body, argNames, compilerOptions, persistOptions);
+        __lenResult = result.length();
+        __result = result.detach();
+    }
+
+} // namespace
+
+MODULE_INIT(INIT_PRIORITY_STANDARD)
+{
+    wasmLanguageHelper::callbacks = std::make_shared<wasmLanguageHelper::Callbacks>();
+    init(wasmLanguageHelper::callbacks);
+    return true;
+}
+
+MODULE_EXIT()
+{
+    kill();
```

Comment:
This could do with a clearer name - hard to tell from the context what it does.  (I didn't search to find it.)